src/db_adapters/receipts.rs

```diff
@@ -69,31 +72,90 @@ async fn store_chunk_receipts(
         .iter()
         .enumerate()
         .filter_map(|(index, r)| {
-            if let Some(transaction_hash) =
-            tx_hashes_for_receipts.get(r.receipt_id.to_string().as_str())
-            {
-                Some(models::Receipt::from_receipt_view(
-                    r,
-                    block_hash,
-                    transaction_hash,
-                    chunk_hash,
-                    index as i32,
-                    block_timestamp,
-                ))
-            } else {
-                warn!(
-                    target: crate::INDEXER_FOR_EXPLORER,
-                    "Skipping Receipt {} as we can't find parent Transaction for it. Happen in block hash {}, chunk hash {}",
-                    r.receipt_id.to_string(),
-                    block_hash,
-                    chunk_hash,
-                );
-                skipping_receipt_ids.insert(r.receipt_id);
-                None
+            // We need to search for parent transaction hash in cache differently
+            // depending on the Receipt kind
+            // In case of Action Receipt we are looking for ReceiptId
+            // In case of Data Receipt we are looking for DataId
+            match r.receipt {
+                near_primitives::views::ReceiptEnumView::Action { .. } => {
+                    if let Some(transaction_hash) =
+                    tx_hashes_for_receipts.get(&crate::ReceiptOrDataId::ReceiptId(r.receipt_id))
```

Comment:
Instead of copy-pasting the whole body, you can match and craft the lookup-key:

```suggestion
            let receipt_or_data_id = match r.receipt {
                near_primitives::views::ReceiptEnumView::Action { .. } => {
                    crate::ReceiptOrDataId::ReceiptId(r.receipt_id)
                }
                near_primitives::views::ReceiptEnumView::Data { data_id, .. } => {
                    crate::ReceiptOrDataId::DataId(data_id)
                }
            };
            if let Some(transaction_hash) = tx_hashes_for_receipts.get(&receipt_or_data_id) {
```